File: DatingApp/API/Helpers/UserParams.cs
using System;

namespace API.Helpers
{
    public class UserParams
    {
        private const int MaxPageSize = 50;
        public int PageNumber { get; set; }
        private int _pageSize = 20;
        public int PageSize
        {
            get => _pageSize; // getter
            set => _pageSize = Math.Min(MaxPageSize, value); // setter
        }
        
        public string CurrentUsename { get; set; }
        public string Gender { get; set; }

        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 150;
        
        public string OrderBy { get; set; } = "lastActive";
    }
}

File: DatingApp/client/src/app/models/user-params.ts
import { User } from './user';
import { Gender } from './Gender.enum';

export class UserParams {
  gender: Gender;
  minAge = 18;
  maxAge = 150;
  pageNumber = 1;
  pageSize = 5;
  orderBy = 'lastActive';

  constructor({ gender }: User) {
    this.gender = gender === Gender.female ? Gender.male : Gender.female;
  }
}
